web/src/main/java/it/pagopa/selfcare/dashboard/web/security/ExchangeTokenServiceV2.java

```diff
@@ -120,7 +120,14 @@ public ExchangedToken exchange(String institutionId, String productId, Optional<
 
         Product product = productsConnector.getProduct(productId);
 
-        environment.ifPresentOrElse(env -> claims.setAudience(product.getBackOfficeEnvironmentConfigurations().get(env).getIdentityTokenAudience())
+        environment.ifPresentOrElse(env -> {
+            if (Objects.isNull(product.getBackOfficeEnvironmentConfigurations()) ||
+                    Objects.isNull(product.getBackOfficeEnvironmentConfigurations().get(env))){
+                        throw new RuntimeException("Invalid Request");
+                    }
+            claims.setAudience(product.getBackOfficeEnvironmentConfigurations().get(env).getIdentityTokenAudience());
+        }
+
```

Comment:
you could reduce complexity by removing the use of Objects.isNull and using Optional for better readability and maintainability. In addition, you could use InvalidRequestException, for ex:
```
environment.ifPresentOrElse(env -> {
    var backOfficeConfigs = product.getBackOfficeEnvironmentConfigurations();
    var envConfig = Optional.ofNullable(backOfficeConfigs)
                            .map(configs -> configs.get(env))
                            .orElseThrow(() -> new InvalidRequestException("Invalid Request"));

    claims.setAudience(envConfig.getIdentityTokenAudience());
});
```